src/patterns/gen-ai/aws-aoss-cw-dashboard/index.ts

```diff
@@ -0,0 +1,374 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import { Duration, Aws } from 'aws-cdk-lib';
+import {
+  Metric,
+  Dashboard,
+  GraphWidget,
+  GraphWidgetView,
+  Stats,
+  TextWidget,
+} from 'aws-cdk-lib/aws-cloudwatch';
+import { CfnCollection } from 'aws-cdk-lib/aws-opensearchserverless';
+import { Construct } from 'constructs';
+
+
+/**
+ * The properties for the CollectionMonitoringProps class.
+ */
+export interface CollectionMonitoringProps {
```

Comment:
Does it support any custom metric prop as well?

Reply:
Not for now, but we will add some in the future